src/hooks/TimeTravel.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from 'react'
+const useTimeTravel = () => {
+    const [date, setDate] = useState([]);
+    const [index, setIndex] = useState(null)
+    const [currentDate, setCurrentDate] = useState('')
+
+
+    useEffect(() => {
+        if(index !== null){
+        setCurrentDate(date[index])
+        return currentDate;
+         }
+    }, [index])
+
+
+const saveDate = (value) => {
+    setCurrentDate(value)
+    setDate((prevState) => [...prevState.slice(0, index + 1), value, ...prevState.slice(index + 1, prevState.length + 1)])
+    setIndex(date.length)
```

Comment:
this should probably be `setIndex(prevIndex => prevIndex + 1)` date.length will not line up with that actual index.